Adds new parser AliasDevice and fast search brands by indexes

sanchezzzhak opened the pull request:
issue #6267

sanchezzzhak committed:
feat(device) create parser detect device hard name form useragent

sanchezzzhak committed:
feat(device) remove brand result alias devices;

sanchezzzhak committed:
feat(device-alias) correct php docs usages;

sanchezzzhak committed:
Merge branch 'master' into 6267

sanchezzzhak committed:
Merge branch 'master' into 6267

sgiehl committed:
Merge branch 'master' into 6267

sanchezzzhak committed:
Merge remote-tracking branch 'upstream/master' into 6267

sanchezzzhak committed:
feat(alias device) adapted for version 4

sanchezzzhak committed:
Merge remote-tracking branch 'origin/6267' into 6267

sanchezzzhak committed:
fix(alias device) fixed php cs

sanchezzzhak commented:
@sgiehl is there a ready setup PHPCS for phpstorm 2020.3 ?

sgiehl commented:
if you installed the composer dependencies there should be one in `vendor\bin\phpcs` (and `vendor\bin\phpcbf` for automatic fixes)

sanchezzzhak committed:
feat(alias device) added support UCBrowser browser

sanchezzzhak committed:
fix(alias device) php cs fixed

sanchezzzhak committed:
fix(alias device) fixed test alias

sanchezzzhak committed:
Merge branch 'master' into 6267

sanchezzzhak committed:
feat(alias device) added detect raw name models for browsers: Dalvik, MQQBrowser, UCWEB
feat(alias device) result empty

sanchezzzhak committed:
feat(alias device) parse result null is not detect

sanchezzzhak committed:
feat(alias device) fix phpcs

sanchezzzhak committed:
Merge branch 'master' into 6267

sanchezzzhak committed:
Merge branch 'master' into 6267

sanchezzzhak committed:
Merge branch 'master' into 6267

sanchezzzhak committed:
feat(alias device) added option to disable brand replacement by empty value
feat(mics) added a new console tool: checks the file and finds useragents that are not present in the tests

sanchezzzhak committed:
Merge branch 'master' into 6267

sanchezzzhak committed:
fix: YAML Lint

mimmi20 commented on Tests/Parser/fixtures/alias_devices.yml:
```diff
@@ -0,0 +1,99 @@
+   name: U5
+
+-
+ user_agent: Mozilla/5.0 (Android 7.0; samsung SM-A710F) AppleWebKit/537.36 KHTML, like Gecko Version/4.0 Chrome/30.0.0.0 Mobile Safari/537.36 SurfBrowser/3.0
```
Please fix indentation

mimmi20 commented on Tests/Parser/AliasDeviceTest.php:
```diff
@@ -0,0 +1,48 @@
+            return;
+        }
+
+        $this->assertEquals(null, $result);
```
Please use `$this->assertNull($result)`

sanchezzzhak committed:
fix: change method check null to assertNull in AliasDeviceTest & formatting of fixture alias_devices.yml

sanchezzzhak committed:
Merge branch 'master' into 6267

sanchezzzhak committed:
feat(alias device) detect models for the browser: ObigoInternetBrowser
feat(alias device) detect models for the browser: InettvBrowser

sanchezzzhak committed:
fix: phpcs & YAML lint

sanchezzzhak committed:
Merge branch 'master' into 6267

sanchezzzhak committed:
feat(alias device) added regex get device model for useragents: WhatsApp, Facebook, Instagram, CFNetwork, UnityPlayer, BlackBerry, Wrong AndroidUA

sanchezzzhak committed:
feat(misc) deleted extra file checkingUseragentFileForTest.php

sanchezzzhak committed:
Merge branch 'master' into 6267

sanchezzzhak committed:
Merge branch 'master' into 6267

sanchezzzhak commented:
I found a good use for AliasDevice, I will try to pass the changes from https://github.com/sanchezzzhak/node-device-detector/tree/pr_device_info8 in PHP

My 22k tests were reduced from 2 minutes to 43 seconds, without cache

sanchezzzhak committed:
feat(alias-device) update detect fixtures

sanchezzzhak committed:
feat(misc) create misc script generate device indexes
feat(parser) make parseAllMatch, parsers

sanchezzzhak committed:
feat(device abstract parser) The code is optimized, from repeated calls.

sanchezzzhak committed:
feat(device abstract parser) generate device indexes

sanchezzzhak committed:
fix: phpcs

sanchezzzhak committed:
fix: phpdoc

sanchezzzhak committed:
fix: phpcs public method move to up

sanchezzzhak committed:
fix: phpcs & rename method parsers to parseResult

sanchezzzhak committed:
fix: phpdocs

sanchezzzhak committed:
fix: method name

sanchezzzhak committed:
fix dump number yaml

sanchezzzhak committed:
feat(parser) move parsers device indexes in main class to save memory

sanchezzzhak committed:
fix: phpcs

sanchezzzhak commented:
summary

Added new methods to device parsers
public list:
AbstractDeviceParser::setBrandIndexes(array brands)  - parsing only selected brands
AbstractDeviceParser::parseAllMatch()    - same as parse, only returns all matches
some checks in the code have been moved as helper methods
AbstractDeviceParser::hasBrand(string brand)
AbstractDeviceParser::hasDeviceType(string deviceType)
private list:
parseResult(bool all = false) - this method implements new scanning logic
parseForBrand
shouldExistBrandOrException

Without indexing, tests are performed for 43 seconds, with 20 seconds.

Where can this be useful?
Mainly in long-lived processes, realtime services, background tasks, testing

sanchezzzhak committed:
Merge remote-tracking branch 'upstream/master' into 6267

sanchezzzhak committed:
feat(device) improve parser AliasDevice for windows phone user-agents

sanchezzzhak committed:
Merge branch 'master' into 6267

sanchezzzhak committed:
update device alias regex and update device indexes

sanchezzzhak commented:
Rejected - this function is not needed yet.
Since there is memcache.

**I don't delete the branch in case it comes in handy.**

In Node Js, this thing works fine.
